lib/rules/yield-effects.js

```diff
@@ -0,0 +1,85 @@
+"use strict";
+
+var Effects = require("../utils/effects");
+
+var knownEffects = Effects.knownEffects();
+
+module.exports = {
+  meta: {
+    docs: {
+      description: "enforce yield in front of effects",
+      category: "Possible Errors",
+      recommended: true,
+    },
+    hasSuggestions: true,
+    schema: [],
+  },
+
+  create: function (context) {
+    var inYieldDepth = 0;
+    var inGeneratorDepth = 0;
+    var effectLocalNames = [];
+    var effectImportedNames = [];
+
+    function enterFunction(node) {
+      if (node.generator) {
+        ++inGeneratorDepth;
+      }
+    }
+    function exitFunction(node) {
+      if (node.generator) {
+        --inGeneratorDepth;
+      }
+    }
+    return {
+      ImportDeclaration: function (node) {
+        if (Effects.isEffectImport(node.source.value)) {
```

Comment:
The logic here seems quite similar to the one in delegate-effects. Can they use the same approach or can the code even be extracted into utils?